src/main/java/net/fabricmc/loom/api/processors/JarProcessor.java

```diff
@@ -22,16 +22,40 @@
  * SOFTWARE.
  */
 
-package net.fabricmc.loom.processors;
+package net.fabricmc.loom.api.processors;
 
-import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
 
 import org.gradle.api.Project;
+import org.objectweb.asm.commons.Remapper;
+
+import net.fabricmc.loom.task.RemapJarTask;
 
 public interface JarProcessor {
```

Comment:
None of the methods should accept `Project`. `processRemapped` also shouldn't accept a gradle task.

Reply:
> None of the methods should accept `Project`.
Why? Even the access widener needs one to get the access widener file from the extension

Reply:
Well they really should be project independent. Even gradle's builtin artifact transfrom is project independent.

Reply:
> Why? Even the access widener needs one to get the access widener file from the extension
 
They can get it from the constructor and store it in a field, there’s no reason to pass a singleton. same for the remapjartask you can get it with project.withType()

Reply:
> same for the remapjartask you can get it with project.withType()

You don't get the exact `RemapJarTask` being used